### ISSUE TITLE
macos-latest(12.7) and PHP 8.1 Library not loaded: libsodium.26.dylib

### ISSUE THREAD

**Issue by Yurunsoft** [shivammathur/setup-php#780]
**Describe the bug**

**Version**
- [ ] I have checked releases, and the bug exists in the latest patch version of `v1` or `v2`.
- [x] `v2`
- [ ] `v1`

**Runners**

- [x] GitHub Hosted
- [ ] Self Hosted

**Operating systems**

macos-latest

**PHP versions**

8.1

**To Reproduce**

```yml
      - name: Setup PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: 8.1
          tools: pecl
          extensions: >
            apcu, bcmath, curl, openssl, mbstring, intl, json, redis, mysqli, pdo, pdo_mysql, sockets, zip, :opcache
```

**Expected behavior**

No problem

**Screenshots/Logs**

![image](https://github.com/shivammathur/setup-php/assets/20104656/cad12cd5-51ce-427c-a7d0-76facf97346a)
```log
dyld[2040]: Library not loaded: '@loader_path/../../../../opt/libsodium/lib/libsodium.26.dylib'
  Referenced from: '/usr/local/Cellar/php@8.1/8.1.[25](https://github.com/imiphp/imi/actions/runs/6780634524/job/18429590036#step:6:26)_1/bin/php'
  Reason: tried: '/usr/local/Cellar/php@8.1/8.1.25_1/bin/../../../../opt/libsodium/lib/libsodium.[26](https://github.com/imiphp/imi/actions/runs/6780634524/job/18429590036#step:6:27).dylib' (no such file), '/usr/local/lib/libsodium.26.dylib' (no such file), '/usr/lib/libsodium.26.dylib' (no such file)
```

**Additional context**

macos-latest(12.6.9) OK
macos-latest(12.7) error

**Are you willing to submit a PR?**

**Comment by SystemKeeper** [comment]
Same happens for PHP 8.0 on macos runner

**Comment by shivammathur** [comment]
@Yurunsoft @SystemKeeper 

Fixed in [`2.27.1`](https://github.com/shivammathur/setup-php/releases/tag/2.27.1).

If you use `v2` no changes are needed, If you pin the version or hash, please upgrade,
Then, please rerun your workflow, it should install correctly now.

**Comment by SystemKeeper** [comment]
Works fine, thank you very much for the quick fix!
